fe/fe-core/src/main/java/com/starrocks/planner/SchemaScanNode.java

```diff
@@ -328,17 +329,29 @@ public void computeFeNodes() {
     }
 
     public void computeBeScanRanges() {
-        for (Backend be : GlobalStateMgr.getCurrentState().getNodeMgr().getClusterInfo().getIdToBackend().values()) {
+        List<ComputeNode> nodeList;
+        if (RunMode.getCurrentRunMode() == RunMode.SHARED_DATA) {
+            long warehouseId = ConnectContext.get().getCurrentWarehouseId();
```

Comment:
‌‌‌‌‌Will this operation fail if the user has not executed "set warehouse = xxx"?

Reply:
report error

Reply:
‌‌‌‌‌‌This could be a problem, especially with the community version.

Reply:
Current users are not aware of warehouses, or the open source version only supports one warehouse. If you choose another warehouse, reporting an error is a reasonable logic